### PR TITLE
handle default address when previous is removed

### PR DESCRIPTION
**Pull request by KacperMekarski**
Solves issue when default address is destroyed, but it is still default user address.

### CODE REVIEW

File: api/spec/requests/spree/api/v2/storefront/account/addresses_spec.rb [hunk 1]
```diff
@@ -49,10 +49,10 @@
     end
 
     context 'when address can not be deleted' do
+      let!(:order) { create(:completed_order_with_totals, bill_address: address) }
```
Comment (KacperMekarski) [inline api/spec/requests/spree/api/v2/storefront/account/addresses_spec.rb:52]: I was getting `address can not be destroyed error`, so I made order undeletable in other way.

File: core/app/models/spree/address.rb [hunk 2]
```diff
@@ -225,5 +226,12 @@ def postal_code_validate
 
       errors.add(:zipcode, :invalid) unless ::ValidatesZipcode.valid?(formatted_zip, country_iso.upcase)
     end
+
+    def assign_new_default_address_to_user
+      return unless user
+
+      user.update(bill_address: user.addresses.last) if user.bill_address == self
+      user.update(ship_address: user.addresses.last) if user.ship_address == self
```
Comment (damianlegawiec) [inline core/app/models/spree/address.rb:234]:
```suggestion
      user.update(bill_address: user.addresses.last) if user.bill_address == self
      user.update(ship_address: user.addresses.last) if user.ship_address == self
```

@KacperMekarski let's not do 2 seperate SQL queries, we can assign attributes to object and do one `.save!`

**Review by damianlegawiec — CHANGES_REQUESTED** [review]
@KacperMekarski one test is failing:

https://app.circleci.com/pipelines/github/spree/spree/3095/workflows/4e0177d8-c899-445b-99de-d4e3f9c105af/jobs/29271

File: core/app/models/spree/address.rb (hunk 2 above)
Comment (KacperMekarski) [reply core/app/models/spree/address.rb]: done!

**Comment by KacperMekarski** [comment]
> @KacperMekarski one test is failing:
> 
> https://app.circleci.com/pipelines/github/spree/spree/3095/workflows/4e0177d8-c899-445b-99de-d4e3f9c105af/jobs/29271

it should pass now.

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**